### DISCUSSION TITLE
Getting error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

### DISCUSSION THREAD

**Discussion by ahbilalkhan** [prisma/prisma#19669] [Q&A] [answered]
Iam setting up prisma on my nodejs express app via docker 

iam getting the following error 

Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
    at new PrismaClient (/app/node_modules/.prisma/client/index.js:3:11)
    at Object.<anonymous> (/app/src/index.ts:10:16)
    at Module._compile (node:internal/modules/cjs/loader:1196:14)
    at Module._compile (/app/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
    at Object.newLoader [as .ts] (/app/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1074:32)
    at Function.Module._load (node:internal/modules/cjs/loader:909:12)
    at Function.runMain (node:internal/modules/run_main:81:12)
    at Object.<anonymous> (/app/node_modules/@babel/node/src/_babel-node.ts:222:12)
    at Module._compile (node:internal/modules/cjs/loader:1196:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
    at Module.load (node:internal/modules/cjs/loader:1074:32)
    at Function.Module._load (node:internal/modules/cjs/loader:909:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:22:47
[nodemon] app crashed - waiting for file changes before starting...

**Comment by ludralph** [comment]
Hi @ahbilalkhan-10p 👋 

Did you run `npx prisma generate`? You said you're using docker, do you have this command `COPY prisma ./prisma/` in your docker file?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ahbilalkhan** [comment]
yes i have added it in my docker file. Here is my docker-compose build logs

Building api
Step 1/13 : FROM node:16
 ---> 4fbbe8e45ea1
Step 2/13 : WORKDIR /app
 ---> Using cache
 ---> 486e44fdfb6f
Step 3/13 : COPY package.json ./
 ---> Using cache
 ---> ccd5e727a7c6
Step 4/13 : COPY prisma ./prisma/
 ---> Using cache
 ---> d3821a4b9ac5
Step 5/13 : RUN npm cache clean --force
 ---> Using cache
 ---> 25d09a4c52fd
Step 6/13 : RUN npm install
 ---> Using cache
 ---> 6fc7451bbbbb
Step 7/13 : COPY . .
 ---> 3267baf8f204
Step 8/13 : COPY ./entrypoint.sh /entrypoint.sh
 ---> fcbf792fb8fe
Step 9/13 : COPY ./wait-for-it.sh /wait-for-it.sh
 ---> 5c487e69465b
Step 10/13 : RUN chmod +x /entrypoint.sh
 ---> Running in af111737b99a
Removing intermediate container af111737b99a
 ---> f50d357c6384
Step 11/13 : RUN chmod +x /wait-for-it.sh
 ---> Running in 8a981370591c
Removing intermediate container 8a981370591c
 ---> d5facedd6f78
Step 12/13 : EXPOSE 3000
 ---> Running in 68fd2a340613
Removing intermediate container 68fd2a340613
 ---> eb20a26b9488
Step 13/13 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Running in dfc4f1a54fe9
Removing intermediate container dfc4f1a54fe9
 ---> f706f595b569
Successfully built f706f595b569
Successfully tagged morris-exports-api_api:latest
Recreating db ... done
Recreating api ... done
Attaching to db, api
db     | 
db     | PostgreSQL Database directory appears to contain a database; Skipping initialization
db     | 
db     | 2023-06-08 09:58:23.778 UTC [1] LOG:  starting PostgreSQL 14.1 on x86_64-pc-linux-musl, compiled by gcc (Alpine 10.3.1_git20211027) 10.3.1 20211027, 64-bit
db     | 2023-06-08 09:58:23.779 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db     | 2023-06-08 09:58:23.779 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db     | 2023-06-08 09:58:23.780 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db     | 2023-06-08 09:58:23.783 UTC [21] LOG:  database system was interrupted; last known up at 2023-06-08 09:55:17 UTC
db     | 2023-06-08 09:58:23.837 UTC [21] LOG:  database system was not properly shut down; automatic recovery in progress
db     | 2023-06-08 09:58:23.838 UTC [21] LOG:  redo starts at 0/1729FF8
db     | 2023-06-08 09:58:23.838 UTC [21] LOG:  invalid record length at 0/172A048: wanted 24, got 0
db     | 2023-06-08 09:58:23.838 UTC [21] LOG:  redo done at 0/1729FF8 system usage: CPU: user: 0.00 s, system: 0.00 s, elapsed: 0.00 s
db     | 2023-06-08 09:58:23.845 UTC [1] LOG:  database system is ready to accept connections
api    | wait-for-it.sh: waiting 15 seconds for localhost:5432
api    | wait-for-it.sh: timeout occurred after waiting 15 seconds for localhost:5432
api    | 
api    | > api@1.0.0 generate
api    | > npx prisma generate
api    | 
api    | Environment variables loaded from .env
api    | Prisma schema loaded from prisma/schema.prisma
api    | 
api    | ✔ Generated Prisma Client (4.15.0 | library) to ./prisma/generated/clientPg in 218ms
api    | You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
api    | ```
api    | import { PrismaClient } from './prisma/generated/clientPg'
api    | const prisma = new PrismaClient()
api    | ```
api    | 
api    | > api@1.0.0 database:deploy
api    | > npx prisma migrate deploy --preview-feature
api    | 
api    | Environment variables loaded from .env
api    | Prisma schema loaded from prisma/schema.prisma
api    | Datasource "db": PostgreSQL database "postgres" at "172.16.16.62:5432"
api    | 
api    | 1 migration found in prisma/migrations
api    | 
api    | 
api    | No pending migrations to apply.
api    | 
api    | > morris-exports-api@1.0.0 start
api    | > nodemon -r dotenv/config
api    | 
api    | [nodemon] 2.0.22
api    | [nodemon] to restart at any time, enter `rs`
api    | [nodemon] watching path(s): src/**/* package-lock.json
api    | [nodemon] watching extensions: ts
api    | [nodemon] starting `babel-node src/index.ts --extensions ".ts" -r dotenv/config`
api    | /app/node_modules/.prisma/client/index.js:3
api    |     throw new Error(
api    |     ^
api    | 
api    | Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
api    | In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
api    |     at new PrismaClient (/app/node_modules/.prisma/client/index.js:3:11)
api    |     at Object.<anonymous> (/app/src/index.ts:10:16)
api    |     at Module._compile (node:internal/modules/cjs/loader:1196:14)
api    |     at Module._compile (/app/node_modules/pirates/lib/index.js:136:24)
api    |     at Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
api    |     at Object.newLoader [as .ts] (/app/node_modules/pirates/lib/index.js:141:7)
api    |     at Module.load (node:internal/modules/cjs/loader:1074:32)
api    |     at Function.Module._load (node:internal/modules/cjs/loader:909:12)
api    |     at Function.runMain (node:internal/modules/run_main:81:12)
api    |     at Object.<anonymous> (/app/node_modules/@babel/node/src/_babel-node.ts:222:12)
api    |     at Module._compile (node:internal/modules/cjs/loader:1196:14)
api    |     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
api    |     at Module.load (node:internal/modules/cjs/loader:1074:32)
api    |     at Function.Module._load (node:internal/modules/cjs/loader:909:12)
api    |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
api    |     at node:internal/main/run_main_module:22:47
api    | [nodemon] app crashed - waiting for file changes before starting...

**Reply by janpio** [reply, marked as answer]
Note how the Client is generated to `./prisma/generated/clientPg` but the error message is thrown by `node_modules/.prisma/client/index.js`.

Are you importing the correct Prisma Client?

**Reply by ahbilalkhan** [reply]
Thanks for the catch @janpio 
I added the wrong import path of PrismaClient 
Changing import path from 
`import { PrismaClient } from '@prisma/client'`
to 
`import { PrismaClient } from '../prisma/generated/clientPg'`

solved the issue

**Reply by inapeace0** [reply]
Awesome solution!!!
I figured out my issue with this solution. Thanks. 🚀

**Reply by andrewjumanca** [reply]
WOW changing the import path solved all my problems. I feel so dumb I didn't realize that sooner.

**Reply by pArmarpaRth22** [reply]
but i have not have location you secified what can i do??
i am deploying my website on render and i am getting prisma is not initialized error

**Reply by wrspada02** [reply]
Why is it a wrong import? It's imported from the node_modules, the package installation. How have you gotten this path? I don't have this path to run. Still figuring it out, running "npx prisma generate" doesn't work as well.

**Reply by pantharshit007** [reply]
> Thanks for the catch @janpio I added the wrong import path of PrismaClient Changing import path from `import { PrismaClient } from '@prisma/client'` to `import { PrismaClient } from '../prisma/generated/clientPg'`
> 
> solved the issue

you have my thanks mate I was stuck on this issue since last night 
I was using this with auto import `import { PrismaClient } from "@prisma/client/extension";`
but then I changed to this `import { PrismaClient } from "@prisma/client/edge";` 
and now it's working

**Reply by tamim360** [reply]
**npx prisma generate** worked for me

**Reply by xuhdev** [reply]
Copy the files `prisma/schema.prisma` to the root of the distribution directory (NOT under `prisma/` sub directory!) works for me. I can keep the following line:

```js
import { PrismaClient } from '@prisma/client'
```

Also, must run `npm install` AFTER the schema file is in place during deployment.

**Reply by efernandes-tech** [reply]
https://www.prisma.io/docs/getting-started/quickstart-sqlite#4-explore-how-to-send-queries-to-your-database-with-prisma-client

```
import { PrismaClient } from '../generated/prisma'
```
or
```
import { PrismaClient } from '@/generated/prisma';
```

**Reply by lakshyarohila** [reply]
> Note how the Client is generated to `./prisma/generated/clientPg` but the error message is thrown by `node_modules/.prisma/client/index.js`.
> 
> Are you importing the correct Prisma Client?

your are great bro i deleted my project 4 time after i find the sol thx

**Reply by huynhnhutanhduy** [reply]
> Thanks for the catch @janpio I added the wrong import path of PrismaClient Changing import path from `import { PrismaClient } from '@prisma/client'` to `import { PrismaClient } from '../prisma/generated/clientPg'`
> 
> solved the issue

Tks bro !!! This is very helpful to me

```
import { PrismaClient } from '@/generated/prisma';
```

**Comment by nasir1196** [comment]
@prisma/client did not initialize yet. Please run "prisma generate" and try to import it again

**Reply by moteeullahazmi** [reply]
where i run prisma generate command

**Reply by vikrantchavan9** [reply]
> where i run prisma generate command

run in backend directory, before starting backend

**Comment by skushagra9** [comment]
I found the solution to this - 
in the package.json for the root directory, under scripts , add
`"postinstall": "prisma generate --schema=<Enter the location to your prisma.schema file>"`

**Comment by rebase-master** [comment]
Spot on @janpio! I noticed I was importing `PrismaClient` like this `import {PrismaClient} from "@prisma/client/extension";`. 
On changing it to: `import {PrismaClient} from "@prisma/client";`, it worked like a charm!

**Reply by Rohitkr711** [reply]
Yup it worked

**Comment by xuhdev** [comment]
Copy the files `prisma/schema.prisma` to the root of the distribution directory (NOT under `prisma/` sub directory!) works for me. No need to change the import path.

Also, must run `npm install` AFTER the schema file is in place during deployment.

**Comment by HangLimbu995** [comment]
When generating Prisma, look at which file directory it is being generated in. Mine was generated in lib, and I imported it from there, and it worked. So, check where it is being generated and import from there.

**Reply by PaulaBurgheleaGithub** [reply]
Thank you! I was going crazy, none of the suggestion helped util I read your. I ran the generate command , got back : ✔ Generated Prisma Client (v6.12.0) to ./app/generated/prisma in 100ms so I updated my seeds.to with import { PrismaClient } from "../app/generated/prisma"; . This finally inserted the seeds in the DB.

**Comment by khanarifak07** [comment]
Step 1:- RUN npx prisma generate in the root directory (it will generate the folder named "generated")
Step 2:- Then IMPORT like this "import { PrismaClient } from "../generated/prisma/client.js";

**Reply by gabrielkramermota** [reply]
🎉

**Reply by dapp-connect** [reply]
God bless you

**Reply by okamixdev** [reply]
oh man life saver!!

**Reply by keval06** [reply]
Thank you brother ❤️

**Reply by ticketapp** [reply]
Not the idiomatic way but the only one that worked for my nextJs app

**Reply by saidMounaim** [reply]
Thank you

**Reply by mskdeveloper** [reply]
i couldn't find folder generate but when you said where is it i found it and i change the import url and finally it worked
thank you bro

**Reply by Danilzone** [reply]
**Thank you very much for your help! You helped me out!** But I'm still wondering, what causes this problem...? I don't quite understand...

**Reply by PaulaBurgheleaGithub** [reply]
so that path changes depending on the project structure, I am using next.js (no src folder) , after running npm i (or for some the generate command) I got back this response ✔ Generated Prisma Client (v6.12.0) to ./app/generated/prisma in 100ms. Then in my seed.ts I've updated the import like so : import { PrismaClient } from "../app/generated/prisma";
So the key is to check what the generate command returns, this really should be outlined better in the docs. Hope this helps

**Reply by MaykMendes** [reply]
Thanks

**Reply by SuperLoveDev** [reply]
mine is not working. Am so tired

**Comment by CuylerD** [comment]
My original import was
  import { PrismaClient } from '@prisma/client';

had to change it to
  import { PrismaClient } from '@/lib/generated/prisma';

**Reply by srivatsajoshi** [reply]
Thanks. It worked

**Reply by kevinlaiGH** [reply]
why does this works @.@
I was banging my head for like 3 hours because of this

**Reply by codecapsules** [reply]
Thanks you !!!

**Comment by afnx** [comment]
I think there is some confusion about how this should be handled. Here is the [docs](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/generating-prisma-client) that explains how you should generate and import the Prisma Client.

It's recommended that you generate the Prisma Client in a custom location. The Prisma Client will no longer be generated in node_modules, so you cannot import using `@prisma/client`.

You need to specify a location in your`schema.prisma` to generate it:

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "app/generated/prisma/client" // Change this path to whatever you want
}
```

Then, make sure you run `npx prisma generate` and import it from the path you previously specified:

```ts
import { PrismaClient } from 'app/generated/prisma/client'
const prisma = new PrismaClient()
```

Also, don't simply copy and paste the import path you chose. Ensure that your file can access the Prisma Client. For example, if you generated the Prisma Client in `'app/generated/prisma/client` and your code is inside the `app` folder, your import path should look like this:

```ts
import { PrismaClient } from 'generated/prisma/client'
```

**Reply by ast5382** [reply]
this helped me, thanks!

**Reply by Olena-Muliarchuk** [reply]
Thank you for sharing your research; it helped me quickly find and localize the problem.

**Comment by jeanRandria** [comment]
change
 import { PrismaClient } from '@prisma/client';
to 
import { PrismaClient } from 'generated/prisma';

**Comment by iamyuu** [comment]
Same as the other, why does docs lie to us?

![3EES6f9E 2025-05-18 at 01 27 PM@2x](https://github.com/user-attachments/assets/be1c3dd2-3d42-4801-bb85-e5ab13af50b7)

**Comment by leoslf** [comment]
try generating the client output to "../node_modules/.prisma/client" instead

```prisma
  generator client {
    provider = "prisma-client-js"
    output   = "../node_modules/.prisma/client"
  }
```

**Reply by DesertCookie** [reply]
This did it for me. It was the first time I installed Prisma for a little while. Just a month ago the default installation would work, but with the very latest version it did not.

**Comment by truong0vanchien** [comment]
I have no ideas how it works. But it's from mine.
![image](https://github.com/user-attachments/assets/12c2e2d8-a559-411e-b206-376365dc1eb8)
![image](https://github.com/user-attachments/assets/b4f6c0c1-4c54-4acb-a165-06ace7afba13)

**Comment by vikasutf8** [comment]
### Its should OPEN again !!! 

```
[nodemon] starting `node ./dist/index.js`
/Users/vikasarya/EndRevist/graphql/GraphQL-/server/node_modules/.prisma/client/default.js:43
    throw new Error('@prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.');
          ^

Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
    at new PrismaClient (/Users/vikasarya/EndRevist/graphql/GraphQL-/server/node_modules/.prisma/client/default.js:43:11)
    at file:///Users/vikasarya/EndRevist/graphql/GraphQL-/server/dist/config/database.js:2:16
    at ModuleJob.run (node:internal/modules/esm/module_job:271:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:578:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)

Node.js v22.14.0

```

this is simple initialization of prisma client
`import { PrismaClient } from "@prisma/client";


const prisma = new PrismaClient();

export default prisma;
`
```

package.json
```
{
  "name": "server",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "generate": "prisma generate",
    "compile": "tsc",
    "start": "npm run compile && node ./dist/index.js",
    "dev": "tsc -w",
    "server": "nodemon ./dist/index.js"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@apollo/server": "^4.12.2",
    "@prisma/client": "^6.10.1",
    "cors": "^2.8.5",
    "dotenv": "^16.5.0",
    "express": "^5.1.0",
    "graphql": "^16.11.0"
  },
  "devDependencies": {
    "@types/node": "^24.0.3",
    "prisma": "^6.10.1",
    "typescript": "^5.8.3"
  }
}
```

As generally as we start integrate Prisma ORM
npm i prisma @prisma/client
npx prisma init -y
npx prisma migrate dev


After all migration file and other generated folders 
npm run dev ==> for tsc -w
npm run server 

!!!ERROR : 
         ^

Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
    at new PrismaClient (/Users/vikasarya/EndRevist/graphql/GraphQL-/server/node_modules/.prisma/client/default.js:43:11)
    at file:///Users/vikasarya/EndRevist/graphql/GraphQL-/server/dist/config/**database.js:2:16**




HOW TO RESOLOVE IT

**Comment by abbaty48** [comment]
In me case, removing the output from `generator client` worked
`generator client {
  provider = "prisma-client-js"
}`

then importing the `PrismaClient` from `@Prisma/Client`

**Reply by vikasutf8** [reply]
Great Advice and It works fine in this case

**Reply by SuperLoveDev** [reply]
i did it but its not working

**Comment by SuperLoveDev** [comment]


**Reply by SuperLoveDev** [reply]
can someone help for me for the dev sake. Got stuck now for 24hrs

[ watch ] build succeeded, watching for changes...                                                 ┃
 └                                                    ┃  Debugger listening on ws://localhost:6001/4d091419-5542-4f61-bdff-caae697a9c16                     ┃
  ✔    @eshop/auth-servic:build:productio       50ms  ┃  Debugger listening on ws://localhost:6001/4d091419-5542-4f61-bdff-caae697a9c16                     ┃
                                                      ┃  For help, see: https://nodejs.org/en/docs/inspector                                                ┃
                                                      ┃                                                                                                     ┃
                                                      ┃  [dotenv@17.2.2] injecting env (0) from .env -- tip: ⚙️e load multiple .env files with { path: ['.   ┃
                                                      ┃  env.local', '.env'] }                                                                              ┃
                                                      ┃                                                                                                     ┃
                                                      ┃                                                                                                     ┃
                                                      ┃  /Users/kaderadenihi/Desktop/eshop/eshop/node_modules/.prisma/client/default.js:43                  ┃
                                                      ┃      throw new Error('@prisma/client did not initialize yet. Please run "prisma generate" and try   ┃
                                                      ┃   to import it again.');                                                                            ┃
                                                      ┃            ^                                                                                        ┃
                                                      ┃  Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it    ┃
                                                      ┃  again.                                                                                             ┃
                                                      ┃      at new PrismaClient (/Users/kaderadenihi/Desktop/eshop/eshop/node_modules/.prisma/client/def   ┃
                                                      ┃  ault.js:43:11)                                                                                     ┃
                                                      ┃      at Object.<anonymous> (/Users/kaderadenihi/Desktop/eshop/eshop/packages/libs/prisma/index.ts   ┃
                                                      ┃  :9:16)                                                                                             ┃
                                                      ┃      at Module._compile (node:internal/modules/cjs/loader:1565:14)                                  ┃
                                                      ┃      at Object..js (node:internal/modules/cjs/loader:1708:10)                                       ┃
                                                      ┃      at Module.load (node:internal/modules/cjs/loader:1318:32)                                      ┃
                                                      ┃      at Function.<anonymous> (node:internal/modules/cjs/loader:1128:12)                             ┃
                                                      ┃      at Function.Module._load (/Users/kaderadenihi/Desktop/eshop/eshop/node_modules/@nx/js/src/ex   ┃
                                                      ┃  ecutors/node/node-with-require-overrides.js:18:31)                                                 ┃
                                                      ┃      at TracingChannel.traceSync (node:diagnostics_channel:322:14)                                  ┃
                                                      ┃      at wrapModuleLoad (node:internal/modules/cjs/loader:219:24)                                    ┃
                                                      ┃      at Module.require (node:internal/modules/cjs/loader:1340:12)                                   ┃
                                                      ┃

**Comment by JuniorCalva40** [comment]
You can use this way 
`generator client {
  provider = "prisma-client-js"
}` and import and the client `import { PrismaClient } from '@prisma/client';`
